From a2552b9f8dce559b2cfc37354c408c86f53c1138 Mon Sep 17 00:00:00 2001
From: Michel Pollet <buserror@gmail.com>
Date: Tue, 22 Jan 2013 14:14:32 +0000
Subject: [PATCH 12/14] regulators: Fix the code to make sure the DT props are
 set

"regulator-type" is required as a property, but it's neither
documented nor tested for validity at runtime, leading to a
crash if you follow the documentation.

Signed-off-by: Michel Pollet <buserror@gmail.com>
---
 Documentation/devicetree/bindings/regulator/gpio-regulator.txt |    2 ++
 drivers/regulator/gpio-regulator.c                             |    4 ++++
 2 files changed, 6 insertions(+)

diff --git a/Documentation/devicetree/bindings/regulator/gpio-regulator.txt b/Documentation/devicetree/bindings/regulator/gpio-regulator.txt
index 63c6598..fba0916 100644
--- a/Documentation/devicetree/bindings/regulator/gpio-regulator.txt
+++ b/Documentation/devicetree/bindings/regulator/gpio-regulator.txt
@@ -4,6 +4,7 @@ Required properties:
 - compatible		: Must be "regulator-gpio".
 - states		: Selection of available voltages and GPIO configs.
                           if there are no states, then use a fixed regulator
+- regulator-type : "voltage" or "current"
 
 Optional properties:
 - enable-gpio		: GPIO to use to enable/disable the regulator.
@@ -22,6 +23,7 @@ Example:
 		regulator-name = "mmci-gpio-supply";
 		regulator-min-microvolt = <1800000>;
 		regulator-max-microvolt = <2600000>;
+		regulator-type = "voltage";
 		regulator-boot-on;
 
 		enable-gpio = <&gpio0 23 0x4>;
diff --git a/drivers/regulator/gpio-regulator.c b/drivers/regulator/gpio-regulator.c
index bae681c..2a66f18 100644
--- a/drivers/regulator/gpio-regulator.c
+++ b/drivers/regulator/gpio-regulator.c
@@ -207,6 +207,10 @@ of_get_gpio_regulator_config(struct device *dev, struct device_node *np)
 
 	of_property_read_string(np, "regulator-type", &regtype);
 
+	if (!regtype) {
+		dev_err(dev, "No 'regulator-type' property found\n");
+		return ERR_PTR(-EINVAL);
+	}
 	if (!strncmp("voltage", regtype, 7))
 		config->type = REGULATOR_VOLTAGE;
 	else if (!strncmp("current", regtype, 7))
-- 
1.7.10.4

